lib/common/xml.c

```diff
@@ -2026,38 +2030,40 @@ pcmk_create_html_node(xmlNode * parent, const char *element_name, const char *id
     return node;
 }
 
-int
-pcmk__element_xpath(const char *prefix, xmlNode *xml, char *buffer,
-                    int offset, size_t buffer_size)
+void
+pcmk__element_xpath(xmlNode *xml, GString *buffer)
 {
-    const char *id = ID(xml);
+    CRM_LOG_ASSERT(buffer != NULL);
```

Comment:
here you'd want to use

    CRM_ASSERT(buffer != NULL);

which will crash the caller (good for catching bugs during development) or

    CRM_CHECK(buffer != NULL, return);

which will log an assertion, dump core, and return from the function (good for things that shouldn't happen but might). CRM_LOG_ASSERT() logs an assertion and dumps core but continues with the code (good for things that indicate something might be wrong if they happen but won't cause problems themselves).

Reply:
Ack. I'll use CRM_ASSERT(). It seems like a null buffer would always be a programmer error for this function.